WordPress/src/main/java/org/wordpress/android/WordPressDB.java

```diff
@@ -278,11 +284,33 @@ public WordPressDB(Context ctx) {
                 // Remove WordPress.com credentials
                 removeDotComCredentials();
                 currentVersion++;
+            case 29:
+                // Migrate WordPress.com token and infos to the DB
+                AccountTable.createTables(db);
+                migratePreferencesToAccountTable(context);
         }
-
         db.setVersion(DATABASE_VERSION);
     }
 
+    private void migratePreferencesToAccountTable(Context context) {
+        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);
+        String oldAccessToken = settings.getString(DEPRECATED_ACCESS_TOKEN_PREFERENCE, null);
+        String oldUsername = settings.getString(DEPRECATED_WPCOM_USERNAME_PREFERENCE, null);
+        Account account = new Account();
```

Comment:
If `oldUsername` is null or empty, it will set a null or empty username, no big deal since the username will be fetched in the `Account.fetchAccountDetails()` call. BTW, I should call `fetchAccountDetails()` just after the migration.